tests/Authentication/Authenticators/SessionAuthenticatorTest.php

```diff
@@ -433,4 +440,43 @@ public function testAttemptUsernameOnly(): void
             'success'    => 1,
         ]);
     }
+
+    /**
+     * Test that any field within the user table can be used as the
+     * login identifier.
+     *
+     * @see https://github.com/codeigniter4/shield/issues/334
+     */
+    public function testAttemptCustomField(): void
+    {
+        // We don't need email, but do need a password set....
+        $this->user->createEmailIdentity([
+            'email'    => '',
```

Comment:
PHPUnit / PHP 7.4 - OCI8 failed. Because Oracle sees `''` as `NULL `.
We have faced this problem before and used the following method to fix it.

https://github.com/kenjis/codeigniter-shield/blob/75c37d0de32937b258156a8819478062111d5f16/src/Models/LoginModel.php#L61-L63

More Info see : #402

Reply:
Oh, great. Thanks!